R/genetic_association.R

```diff
@@ -49,92 +61,48 @@ getGeneticAssociations = function(
   checkDiseaseVariantMap(diseaseVariantMap, scores, genotypes)
   checkGlmFormula(glmFormula, demos)
   modelType = match.arg(modelType)
+  assertNumber(level, lower = 0.5, upper = 1 - .Machine$double.eps)
   assertFlag(dopar)
 
   lmInput = merge(scores, demos, by = 'person_id')
   lmInput[, person_id := as.character(person_id)]
 
   reg = foreach::getDoParRegistered()
   doOp = if (dopar && reg) `%dopar%` else `%do%`
-  foe = foreach(diseaseId = unique(diseaseVariantMap$disease_id), .combine = rbind)
+  foe = foreach(
+    diseaseId = unique(diseaseVariantMap$disease_id), .combine = rbind)
 
   statsAll = doOp(foe, {
 
     lmInputSub = lmInput[disease_id == diseaseId, !'disease_id']
 
-    snpSub = unique(diseaseVariantMap[disease_id == diseaseId]$vid)
+    snpNow = unique(diseaseVariantMap[disease_id == diseaseId]$vid)
+    genoCount = apply(
+      genotypes[, snpNow, drop = FALSE], 2L, uniqueN, na.rm = TRUE)
+    snpSub = snpNow[genoCount > 1]
 
     genotypesSub = data.table(
-      'person_id' = rownames(genotypes),
-      genotypes[, snpSub, drop = FALSE])
-
-    # exclude variants with only one genotype in the population
-    genoCount = melt(
-      genotypesSub[, lapply(.SD, uniqueN, na.rm = TRUE), .SDcols = snpSub],
-      measure.vars = snpSub, variable.name = 'snp', value.name = 'count',
-      variable.factor = FALSE)
-    snpSub = unique(genoCount[count != 1]$snp)
+      person_id = rownames(genotypes), genotypes[, snpSub, drop = FALSE])
 
+    statsSnps = foreach(snp = snpSub, .combine = rbind) %do% {
```

Comment:
I'm not sure about this, but couldn't you just remove `statsSnps = ` and the output of the inner foreach would directly pipe to the output of the outer foreach?

Reply:
Yes, but it makes development easier to explicitly assign the result to a variable.